hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/AbstractEmbeddableMapping.java

```diff
@@ -180,18 +180,25 @@ else if ( attributeMapping instanceof ToOneAttributeMapping ) {
 						declaringType,
 						declaringTableGroupProducer
 				);
-				final int offset = currentIndex;
 				toOne.setIdentifyingColumnsTableExpression(
-						selectableMappings.getSelectable( offset ).getContainingTableExpression()
-				);
-				toOne.setForeignKeyDescriptor(
-						foreignKeyDescriptor.withKeySelectionMapping(
-								declaringType,
-								declaringTableGroupProducer,
-								index -> selectableMappings.getSelectable( offset + index ),
-								creationProcess
-						)
+						selectableMappings.getSelectable( currentIndex ).getContainingTableExpression()
 				);
+				if ( original.getSideNature() == ForeignKeyDescriptor.Nature.KEY ) {
+					final int offset = currentIndex;
+					toOne.setForeignKeyDescriptor(
+							foreignKeyDescriptor.withKeySelectionMapping(
+									declaringType,
+									declaringTableGroupProducer,
+									index -> selectableMappings.getSelectable( offset + index ),
+									creationProcess
+							)
+					);
+				}
+				else {
+					// A toOne may be included in a VirtualIdEmbeddable even if its nature is TARGET,
+					// since it will still be part of the composite identifier
+					toOne.setForeignKeyDescriptor( foreignKeyDescriptor );
```

Comment:
I'm having a hard time understanding why the selection mappings are not applied in this case.

Reply:
Because they are from the key side. This being the target side, we don't need to apply them since they are already correct - but we still need to set the fk descriptor to corretly initialize `toOne`'s properties.